online-shopping-website/src/App.js

```diff
@@ -1,15 +1,54 @@
 import './App.css';
-import { ProductPage } from './Components/ProductPage';
+import {ProductPage} from './Components/ProductPage';
+import {ProductPreview} from './Components/ProductGrid';
+import {ProductDetails} from './Components/ProductDetails';
+import ReactDOM from "react-dom";
+import {BrowserRouter, Outlet, Link, Routes, Route} from "react-router-dom";
 
-function App() {
+export default function App() {
     return (
-        <div className="AppContent">
-            <h1>
-                Bobble
-            </h1>
-            <ProductPage />
-        </div>
+        <BrowserRouter>
+            <Routes>
+                <Route path="/" element={<Layout/>}>
+                    <Route index element={<ProductPage/>}/>
+                    {/*Sample routing, can add any additional necessary pages here*/}
+                    <Route path="/samplePage" element={<SamplePage/>}/>
+                    <Route path="/:productName" element={<ProductDetails/>}/>
```

Comment:
Can we use product IDs instead of names?
If we still want the product name to show up in the url, I'm pretty sure you can mask the url on the client side, so they see the product name, but we see the product id... This isn't necessary anyways.
Maybe a better way would be to have the name ***and*** the id in the url?
We just want to avoid the case where two products may have the same name and the url would link to the wrong product.

Reply:
Will do 👍 

Leaving it for sprint 4 refactoring since I'll be fixing the map() function to a forEach() anyways in the same code block.